app/preload/bookmarks.js

```diff
@@ -0,0 +1,254 @@
+/*
+* Vieb - Vim Inspired Electron Browser
+* Copyright (C) 2019-2025 Jelmer van Arnhem
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+"use strict"
+
+/**
+ * @typedef {{
+ *   bookmarks: Bookmark[],
+ *   folders: {path: string}[],
+ *   lastId: number,
+ *   tags: object[]
+ * }} BookmarkData
+ */
+
+/**
+ * @typedef {{
+ *  children: FolderNode[],
+ *  folderName: string,
+ *  fullpath: string
+ * }} FolderNode
+ */
+
+/**
+ * @typedef {{
+ *  id: number,
+ *  name: string,
+ *  title: string,
+ *  url: string,
+ *  path: string,
+ *  tag: string[],
+ *  keywords: string[],
+ *  bg?: string,
+ *  fg?: string
+ * }} Bookmark
+ */
+
+const {ipcRenderer} = require("electron")
+
+/** @type {FolderNode} */
+const tree = {
+    "children": [],
+    "folderName": "/",
+    "fullpath": "/"
+}
+
+/**
+ * Trigger send to host bookmark data request.
+ */
+const update = () => ipcRenderer.sendToHost("bookmark-data-request")
+
+/**
+ * Trigger bookmark's folder delete.
+ * @param {PointerEvent} e - HTML onclick event.
+ */
+const deleteFolderClick = e => {
+    e.stopPropagation()
+    // @ts-ignore
+    const path = e?.target?.dataset?.path
+    ipcRenderer.sendToHost("delete-folder", path)
+    window.location.reload()
```

Comment:
I would prefer to clear the list and fetch it again, while also showing the result of this action.